zhaquirks/tuya/__init__.py

```diff
@@ -60,13 +93,71 @@ class Command(t.Struct):
         function: t.uint8_t
         data: Data
 
-    manufacturer_server_commands = {0x0000: ("set_data", (Command,), False)}
+    """ Time sync command (It's transparent between MCU and server)
+            Time request device -> server
+               payloadSize = 0
+            Set time, server -> device
+               payloadSize, should be always 8
+               payload[0-3] - UTC timestamp (big endian)
+               payload[4-7] - Local timestamp (big endian)
+
+            Zigbee payload is very similar to the UART payload which is described here: https://developer.tuya.com/en/docs/iot/device-development/access-mode-mcu/zigbee-general-solution/tuya-zigbee-module-uart-communication-protocol/tuya-zigbee-module-uart-communication-protocol?id=K9ear5khsqoty#title-10-Time%20synchronization
+
+            Some devices need the timestamp in seconds from 1/1/1970 and others in seconds from 1/1/2000.
+
+            NOTE: You need to wait for time request before setting it. You can't set time without request."""
+
+    manufacturer_server_commands = {
+        0x0000: ("set_data", (Command,), False),
+        0x0024: ("set_time", (TuyaTimePayload,), False),
+    }
 
     manufacturer_client_commands = {
         0x0001: ("get_data", (Command,), True),
         0x0002: ("set_data_response", (Command,), True),
+        0x0024: ("set_time_request", (TuyaTimePayload,), True),
     }
 
+    def handle_cluster_request(self, tsn: int, command_id: int, args: Tuple) -> None:
+        """Handle time request."""
+
+        if command_id != 0x0024 or self.set_time_offset == 0:
+            return super().handle_cluster_request(tsn, command_id, args)
+
+        # Send default response because the MCU expects it
+        schema = foundation.COMMANDS[foundation.Command.Default_Response][0]
+        self.create_catching_task(
+            self.reply(
+                True, foundation.Command.Default_Response, schema, command_id, tsn=tsn
+            )
+        )
```

Comment:
With `zigpy==0.32.0` you could do 

```suggestion
        # Send default response because the MCU expects it
        self.send_default_rsp(hdr)
```

https://github.com/zigpy/zigpy/blob/2a851f1ce7694c4b331a6a87d5334969d95f57aa/zigpy/zcl/__init__.py#L620